### PR TITLE
Moved code to a new branch

### LINKED ISSUE #23449: Filter does not work on the image selector for image fields
### Describe the bug
The filter search box in the image selector for image fields does not function. You can type into it but the results do not change.

Reproduced on: 22.03.2, 22.10.1

### To Reproduce
1. Open the image selector on an image field
2. Attempt to filter using the filter box

The results in the window will not change. 

### Expected behavior
The filter should work

### Acceptance criteria

- [ ]  We need to also be able to filter by folder
- [ ]  We need to show file assets and dotassets
- [ ]  Make dotassets be able to filter by folders as well (folder field)

### PR DESCRIPTION
**Pull request by hassan-mustafa-baig**
### Proposed Changes
* Added a support to fetch content from ES after searching database. This is a special scenario when the customer performs upgrade from release 21.06 to the release 23.03 then the content must be searched on both the Database and the ElasticSearch

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

image
image
image
image
image

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 1]
```diff
@@ -268,25 +279,42 @@ private Tuple2<String, List<Object>> selectQuery(final BrowserQuery browserQuery
...
+            luceneQuery.append("+baseType:(" + String.join(" OR ", baseTypes)+ ") ");
         }
         if (browserQuery.languageId > 0) {
             sqlQuery.append(" and cvi.lang = ? ");
             parameters.add(browserQuery.languageId);
+            luceneQuery.append("+languageId:" + browserQuery.languageId + " ");
         }
         if (browserQuery.site != null) {
-            sqlQuery.append(" and id.host_inode = ? ");
+            sqlQuery.append(" and (id.host_inode = ? OR id.host_inode = 'SYSTEM_HOST') ");
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:308]: Replace the `'SYSTEM_HOST'` String with our global constant: `Host.SYSTEM_HOST`

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 2]
```diff
@@ -268,25 +279,42 @@ private Tuple2<String, List<Object>> selectQuery(final BrowserQuery browserQuery
...
             parameters.add(browserQuery.site.getIdentifier());
+            luceneQuery.append("+conHost:(SYSTEM_HOST OR " + browserQuery.site.getIdentifier() + ") ");
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:310]: Same thing here.

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 3]
```diff
@@ -54,7 +58,7 @@ public class BrowserAPIImpl implements BrowserAPI {
     private final FolderAPI folderAPI = APILocator.getFolderAPI();
     private final PermissionAPI permissionAPI = APILocator.getPermissionAPI();
     private final ShortyIdAPI shortyIdAPI = APILocator.getShortyAPI();
-
+    private final ESContentletAPIImpl esContentletAPI = new ESContentletAPIImpl();
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:61]: Use the `APILocator.getContentletAPI()` instead.

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java (hunk 3 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:61]: `APILocator.getContentletAPI()` will not work here

image

Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java]: by the way I have added this method in the APILocator.java
image

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/APILocator.java [hunk 4]
```diff
@@ -416,6 +416,15 @@ public static ContentletAPI getContentletAPI() {
...
 
+	/**
+	 * This is the contentletAPI which will be used to perform ES related operations
+	 * normal {@link ContentletAPI} logic.
+	 *
+	 * @return The {@link ContentletAPI} class.
+	 */
+	public static ESContentletAPIImpl getESContentletAPIImpl()
+	{return (ESContentletAPIImpl)getInstance(APIIndex.CONTENTLET_API);}
+
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/APILocator.java:427]: This method is not necessary. You just need to call the already existing `getContentletAPI()`

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/APILocator.java (hunk 4 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotmarketing/business/APILocator.java]: this has been addressed

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPITest.java [hunk 5]
```diff
@@ -484,10 +484,7 @@ public void test_findFileAssetsByHost_workingFileAssets_success()
         }
 
         List<FileAsset> assets = APILocator.getFileAssetAPI().findFileAssetsByHost(site,user,false,true,false,false);
-        assertEquals(fileAssetSize,assets.size());
-        assets.forEach(a-> {
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPITest.java:488]: why did you remove the foreach check?

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPITest.java [hunk 6]
```diff
@@ -516,10 +513,7 @@ public void test_findFileAssetsByHost_archivedFileAssets_success()
         }
 
         List<FileAsset> assets = APILocator.getFileAssetAPI().findFileAssetsByHost(site,user,false,false,true,false);
-        assertEquals(fileAssetSize,assets.size());
-        assets.forEach(a-> {
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPITest.java:520]: same doubt here

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 7]
```diff
@@ -268,25 +276,42 @@ private Tuple2<String, List<Object>> selectQuery(final BrowserQuery browserQuery
...
         if (browserQuery.languageId > 0) {
             sqlQuery.append(" and cvi.lang = ? ");
             parameters.add(browserQuery.languageId);
+            luceneQuery.append("+languageId:" + browserQuery.languageId + " ");
         }
         if (browserQuery.site != null) {
-            sqlQuery.append(" and id.host_inode = ? ");
+            sqlQuery.append(" and (id.host_inode = ? OR id.host_inode = '"+ Host.SYSTEM_HOST +"') ");
             parameters.add(browserQuery.site.getIdentifier());
+            luceneQuery.append("+conHost:("+ Host.SYSTEM_HOST +" OR " + browserQuery.site.getIdentifier() + ") ");
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:307]: It seems like you're not getting the most out of using a StringBuilder because appending many strings with `+` instead of using `.append` makes string management expensive

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 8]
```diff
@@ -268,25 +276,42 @@ private Tuple2<String, List<Object>> selectQuery(final BrowserQuery browserQuery
         final StringBuilder sqlQuery = new StringBuilder("select cvi." + workingLiveInode + " as inode "
                 + " from contentlet_version_info cvi, identifier id, structure struc, contentlet c "
                 + " where cvi.identifier = id.id and struc.velocity_var_name = id.asset_subtype and  "
-                + " c.inode = cvi." + workingLiveInode);
+                + " c.inode = cvi." + workingLiveInode + " and cvi.variant_id='"+DEFAULT_VARIANT.name()+"' ");
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:279]: I might be wrong, but the BrowserAPITest doesn't seem to be covering all the scenarios implemented on this logic where you force the ES search if the record doesn't exist in database

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java (hunk 7 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java]: you are right! @jcastro-dotcms made some improvements in the code

File: dotCMS/.attach_pid200623
[the remark this answers is not in the document]
Comment (hassan-mustafa-baig) [reply dotCMS/.attach_pid200623:1]: Done
